addons/mass_mailing_sms/tests/test_mailing_internals.py

```diff
@@ -59,3 +59,32 @@ def test_mailing_computed_fields(self):
             'contact_list_ids': [(4, self.mailing_list_1.id), (4, self.mailing_list_2.id)]
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('list_ids', 'in', (self.mailing_list_1 | self.mailing_list_2).ids)])
+
+    @users('user_marketing')
+    def test_mailing_get_chars_counts(self):
+        """Test the extra chars counts for estimating total SMS characters length.
+
+          * links are transformed into "[base_url]/r/[link_tracker_code]/s/[sms_id]".
+          * opt-out link is formatted as "\nSTOP SMS : [base_url]/sms/[mailing_id]/[trace_code]".
+        See `get_extra_characters_counts` for safety characters.
+        """
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing',
+            'subject': 'Test',
+            'mailing_type': 'sms',
+            'body_plaintext': 'Coucou hibou',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+        })
+        base_url = self.env['mailing.mailing'].get_base_url()
+
+        expected = {
+            'linkCharsCount': len(f'{base_url}/r//s') + 9,  # 5 for sms id, 4 for tracker_code and safety
+            'optOutLinkCharsCount': len(f"\nSTOP SMS : {base_url}/sms/{mailing.id}/{'x' * self.env['mailing.trace'].CODE_SIZE}"),
+        }
+        self.assertDictEqual(mailing.get_extra_characters_counts(), expected)
+
+        no_mailing = self.env['mailing.mailing']
+        self.assertDictEqual(no_mailing.get_extra_characters_counts(), expected)
+
+        new_mailing = self.env['mailing.mailing'].new()
+        self.assertDictEqual(new_mailing.get_extra_characters_counts(), expected)
```

Comment:
Isn't it valid only because we have few SMS in test db ? I know this won't happen often, but we could try to check we have less than 100K sms